### PR TITLE
[DDW-589] Refactor consolidation status screen epoch calculation

### PR DESCRIPTION
**Pull request by daniloprates**
This PR refactors consolidation status screen epoch calculation to fetching the `slotId` from the API.

## Todo:

- [x] Retrieve the `slotId` from the API
- [x] Use [cardanoexplorer](https://cardanoexplorer.com/api/blocks/pages) as a fallback
  - [x] Implement `cardanoexplorer` fallback for all evironments
- [x] Change the current calculation in the Block Consolidation screen
- [x] Remove local `startTime` variable
- [x] Implement acceptance test for fallback epoch calculation

## Testing

[Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1553156138215000)

We are now fetching the `currentEpoch` number via API. Sometimes this information won't be available for a few milliseconds, when the page should be in its "loading" state (see first screenshot): 

* The numbers in parathesis in the explanation text should be hidden.
* The "x **of y** epochs consolidated" should be hidden.
* The progress bar should not display any text and should have a background animation, showing it's in "loading" mode.

### Testing the "loading state" on a normal situation (API works correctly)

1. When the app is loaded, open the Block Consolidation page;
2. Reload the app (Ctrl+R);
3. The page should be displayed in "loading state" for a few milliseconds;
4. The page should display all the information once the `currentEpoch` is fetched from the API.

### Testing the "loading state" when the API fails

Sometimes the API might fail to retrieve the `currentEpoch`, so as a fallback we fetch just once this information from the [cardanoexplorer api](https://cardanoexplorer.com/api/blocks/pages).

To test it:
1. Run daeadalus on [`staging` env](https://github.com/input-output-hk/daedalus#connect-to-staging-cluster)
2. Comment out [NetworkStatusStore.js#L415](https://github.com/input-output-hk/daedalus/blob/e8a8ae57dbe6b25300f1e375e9486b924f0f1b25/source/renderer/app/stores/NetworkStatusStore.js#L415): `// this.currentEpoch = slotId.epoch;`
3. Open the Block Consolidation screen;
4. The page should be displayed in "loading state" for a few milliseconds;
5. Upon receiving the `currentEpoch`, the page should be displayed fully (second screenshot);
6. On development, the page would remain without the `currentEpoch` information.

### Without the `currentEpoch` information:

image

### With the `currentEpoch` information:

bc

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

**Comment by nikolaglumac** [comment]
@daniloprates @DominikGuzei @MarcusHurney please *DO NOT* merge this one yet. I have to check something first...

**Comment by nikolaglumac** [comment]
@DominikGuzei this is the part which I am worried about... This screen should work regardless of the connection status 🤔 
Not sure what we should do - perhaps we should keep the old code and use it when the connection is unavailable?

**Comment by nikolaglumac** [comment]
@daniloprates I like your changes - the only question is what numer for current epoch we show in case the API is unreachable?

**Comment by nikolaglumac** [comment]
@daniloprates what about system time? Do we still need to fetch it for development/demo cluster?

**Comment by nikolaglumac** [comment]
@daniloprates - @darko-mijic and I have discussed this and we have agreed that we should keep our current calculation of the current epoch and use it when the API info is unavailable so that it is a fallback info.

**Comment by nikolaglumac** [comment]
@daniloprates I have just merged @MarcusHurney PR which adds an acceptance test for the consolidation screen. Can you please merge those changes in and add a test which covers the fallback epoch calculation? 🙏

**Comment by nikolaglumac** [comment]
@daniloprates I have merged the latest develop - the last thing we need to do is to add the acceptance test for the fallback epoch calculation (added a TODO in the PR's description).

**Comment by daniloprates** [comment]
> @daniloprates I have merged the latest develop - the last thing we need to do is to add the acceptance test for the fallback epoch calculation (added a TODO in the PR's description).

@nikolaglumac ✅

**Comment by daniloprates** [comment]
@nikolaglumac the method was using a is not working, currently:

```gherkin
When I inject fault named "FInjIgnoreAPI"
```
I'll try to find another way to test the faulty API fallback.

**Comment by nikolaglumac** [comment]
@daniloprates ping me if you want to discuss this.

**Comment by nikolaglumac** [comment]
@daniloprates can you please change the text on the button from "Support" into "Learn more"? 🙏

**Comment by nikolaglumac** [comment]
@daniloprates during Node-restart on Linux only we get a zero shown on the progress bar:

screenshot_from_2019-03-29_10-17-30

cc @IuliaDolishniak

File: source/main/utils/setupLogging.js [hunk 1]
```diff
@@ -17,13 +17,13 @@ import type {
 } from '../../common/types/logging.types';
 
 const isTest = process.env.NODE_ENV === 'test';
-const isDev = process.env.NODE_ENV === 'development';
+// const isDev = process.env.NODE_ENV === 'development';
```
Comment (nikolaglumac) [inline source/main/utils/setupLogging.js]: @daniloprates can you please revert this change?

File: source/main/utils/setupLogging.js [hunk 2]
```diff
@@ -17,13 +17,13 @@ import type {
...
 
 export const setupLogging = () => {
   const logFilePath = path.join(pubLogsFolderPath, `${APP_NAME}.json.log`);
   ensureDirectoryExists(pubLogsFolderPath);
   log.transports.console.level = isTest ? 'error' : 'info';
-  log.transports.rendererConsole.level = isDev ? 'info' : 'error';
+  log.transports.rendererConsole.level = 'info';
```
Comment (nikolaglumac) [inline source/main/utils/setupLogging.js]: @daniloprates can you please revert this change?

File: source/renderer/app/stores/BlockConsolidationStore.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
+// @flow
+import { observable, action, runInAction } from 'mobx';
+
```
Comment (nikolaglumac) [inline source/renderer/app/stores/BlockConsolidationStore.js]: @daniloprates please remove this blank line 🙏

File: source/renderer/app/stores/BlockConsolidationStore.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import { getNumberOfEpochsConsolidatedChannel } from '../ipc/getNumberOfEpochsConsolidatedChannel';
+import type { GetConsolidatedEpochsCountMainResponse } from '../../../common/ipc/api';
+import type {
+  GetNodeSettingsResponse,
+  GetCurrentEpochFallbackResponse,
+} from '../api/nodes/types';
+
+export default class BlockConsolidationStore extends Store {
+  // Initialize store observables
+  @observable epochsConsolidated: number = 0; // Got from the IPC channel
```
Comment (nikolaglumac) [inline source/renderer/app/stores/BlockConsolidationStore.js]: @daniloprates please change comment for this and the following line - `Got` ->  `Received` 🙏

File: features/step_definitions/block-consolidation-page-steps.js [hunk 5]
```diff
@@ -42,21 +44,28 @@ Then(
...
+        ? ` (${Math.max(currentEpoch - 1, 0)})`
+        : '';
+    }
+
     expect(explanationText).to.equal(
-      `Blocks for the current epoch (${currentEpoch}) and the previous epoch (${Math.max(
-        currentEpoch - 1,
-        0
-      )}) are stored as one file per block. All previous epochs will be consolidated to two files per epoch.`
+      `Blocks for the current epoch${currentEpochText} and the previous epoch${previousEpochText} are stored as one file per block. All previous epochs will be consolidated to two files per epoch.`
```
Comment (nikolaglumac) [inline features/step_definitions/block-consolidation-page-steps.js]: @daniloprates please don't hardcode the content here - we should always match it to the value in the translation file. You can see how we did it here: https://github.com/input-output-hk/daedalus/blob/develop/features/step_definitions/wallets-steps.js#L570

File: source/renderer/app/api/api.js [hunk 6]
```diff
@@ -978,6 +1019,7 @@ export default class AdaApi {
   setSubscriptionStatus: Function;
   setLocalBlockHeight: Function;
   setNetworkBlockHeight: Function;
+  setFaultyNodeSettingsApi: boolean;
```
Comment (nikolaglumac) [inline source/renderer/app/api/api.js]: @daniloprates we should not have this test call defined in the main Api.
Can you please move it to the patchAdaApi? Also I think it is better to use const in the patchAdaApi like we did for this case: https://github.com/input-output-hk/daedalus/blob/develop/source/renderer/app/api/utils/patchAdaApi.js#L119-L121

Comment (daniloprates) [reply source/renderer/app/api/api.js]: @nikolaglumac  `setFaultyNodeSettingsApi` is defined and used [in the patchAdaApi](https://github.com/input-output-hk/daedalus/pull/1339/files/749277390a1c0b65d592cd1cfd54b690d3a1ee89#diff-298cb1b5c528df77cb1c416241432d87R146), but I couldn't find another way of declaring its type. I've done like other methods from the patchApi, like `setSubscriptionStatus`.

Comment (nikolaglumac) [reply source/renderer/app/api/api.js]: Ooooh that's right - I have missed it sorry! You are right :)

**Comment by daniloprates** [comment]
> @daniloprates the improvements look great! I have left only 2 small change requests.
> The bad part is that I see the new test case always failing :(
> Hope you can fix that! 🙏
> 
> Screen Shot 2019-04-01 at 12 46 53

@nikolaglumac tests are running smoothly, now
